### ISSUE TITLE
[Bug] Travis CI build pipeline is broken

### ISSUE THREAD

**Issue by bnoazx005** [bnoazx005/TDEngine2#42]
#### Description

```
In file included from /home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IBaseObject.h:11:0,

                 from /home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/CBaseObject.h:10,

                 from /home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/CBaseObject.cpp:1:

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/./../utils/Utils.h:499:14: error: explicit specialization in non-namespace scope ‘class TDEngine2::CStringUtils’

    template <>

              ^

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/./../utils/Utils.h:506:32: error: too many template-parameter-lists

    TDE2_API static std::string ToString(T&& arg)

                                ^~~~~~~~

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/./../utils/Utils.h:516:25: error: too many template-parameter-lists

    TDE2_API static void _convertToStringsArray(std::array<std::string, size>& outArray) {}

                         ^~~~~~~~~~~~~~~~~~~~~~

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/./../utils/Utils.h:519:25: error: too many template-parameter-lists

    TDE2_API static void _convertToStringsArray(std::array<std::string, size>& outArray, Head&& firstArg, Tail... rest)

                         ^~~~~~~~~~~~~~~~~~~~~~

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/./../utils/Utils.h: In static member function ‘static std::string TDEngine2::CStringUtils::Format(const string&, TArgs ...)’:

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/./../utils/Utils.h:487:29: error: there are no arguments to ‘ToString’ that depend on a template parameter, so a declaration of ‘ToString’ must be available [-fpermissive]

      currArgPattern = "{" + ToString(i) + "}";

                             ^~~~~~~~

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/./../utils/Utils.h:487:29: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)

make[2]: *** [TDEngine2/CMakeFiles/TDEngine2.dir/source/core/CBaseObject.cpp.o] Error 1

make[1]: *** [TDEngine2/CMakeFiles/TDEngine2.dir/all] Error 2

make: *** [all] Error 2

The command "cmake --build ." exited with 2.
```

**Comment by bnoazx005** [comment]
Update:
```
In file included from /home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/CDefaultEngineCoreBuilder.cpp:3:0:

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IPlugin.h:52:147: warning: ‘cdecl’ attribute ignored [-Wattributes]

  typedef TDE2_API TDEngine2::IPlugin* (TDE2_APIENTRY *TCreatePluginCallback)(TDEngine2::IEngineCore* pEngineCore, TDEngine2::E_RESULT_CODE& result);

                                                                                                                                                   ^

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IPlugin.h:52:147: warning: ‘visibility’ attribute ignored [-Wattributes]

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IPlugin.h:66:137: warning: ‘cdecl’ attribute ignored [-Wattributes]

 extern "C" TDE2_API TDEngine2::IPlugin* TDE2_APIENTRY CreatePlugin(TDEngine2::IEngineCore* pEngineCore, TDEngine2::E_RESULT_CODE& result);

                                                                                                                                         ^

In file included from /home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/CBaseJobManager.h:10:0,

                 from /home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/CDefaultEngineCoreBuilder.cpp:6:

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IJobManager.h: In member function ‘TDEngine2::E_RESULT_CODE TDEngine2::IJobManager::SubmitJob(std::function<void(TArgs ...)>, TArgs ...)’:

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IJobManager.h:105:23: error: ‘make_unique’ is not a member of ‘std’

     return _submitJob(std::make_unique<TJob<TArgs...>>(jobCallback, std::forward<TArgs>(args)...));

                       ^

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IJobManager.h:105:53: error: expected primary-expression before ‘>’ token

     return _submitJob(std::make_unique<TJob<TArgs...>>(jobCallback, std::forward<TArgs>(args)...));

                                                     ^

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IJobManager.h:105:94: error: expected ‘)’ before ‘...’ token

     return _submitJob(std::make_unique<TJob<TArgs...>>(jobCallback, std::forward<TArgs>(args)...));

                                                                                              ^

/home/travis/build/bnoazx005/TDEngine2/TDEngine2/source/core/./../../include/core/IJobManager.h:105:99: error: expected ‘)’ before ‘;’ token

     return _submitJob(std::make_unique<TJob<TArgs...>>(jobCallback, std::forward<TArgs>(args)...));

                                                                                                   ^

make[2]: *** [TDEngine2/CMakeFiles/TDEngine2.dir/source/core/CDefaultEngineCoreBuilder.cpp.o] Error 1

make[1]: *** [TDEngine2/CMakeFiles/TDEngine2.dir/all] Error 2

make: *** [all] Error 2
```

**Comment by bnoazx005** [comment]
Fixed in 7e94881e9006e7184b3224155e522f63fc96357d
